### PR TITLE
Adds HERE planner to route package and explicit Zip5 lookups

### PR DESCRIPTION
**Pull request by ntwyman**
## Description

Turns out the HERE API appears to be less expensive and easier to license than BING, so I have moved us over to that for now.

In addition I have added explicit routing for Zip5 to Zip5.

## Reviewer Notes

Assuming you have the HERE environment variables set, then you can `make server_test_all` to run the full tests which hit the HERE servers and check distances returned.

## Code Review Verification Steps

* [x] All tests pass.
* [x] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [n/a] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [n/a] There are no [aXe](https://www.deque.com/products/aXe/) warnings for UI.
* [n/a] This works in IE.
* [n/a] Any new third party client side dependencies (e.g., login.gov, google analytics, CDN libraries, etc.) have been communicated to @willowbl00.
* [*] Request review from a member of a different team.
* [n/a] (TRIAL) Have the Pivotal acceptance criteria been met for this change?

### CODE REVIEW

File: pkg/route/here_planner.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+	latLongResponse.address = address
+
+	// Look up address
+	query := fmt.Sprintf("%s&searchtext=%s", p.geocodeEndPointWithKeys, urlencodeAddress(address))
+	resp, err := p.httpClient.Get(query)
+	if err != nil {
+		p.logger.Error("Getting response from HERE.", zap.Error(err), zap.Object("address", address))
+		latLongResponse.err = errors.Wrap(err, "calling HERE")
+	} else if resp.StatusCode != 200 {
+		p.logger.Info("Got non-200 response from Bing.", zap.Int("http_status", resp.StatusCode), zap.Object("address", address))
```
Comment (macrael) [inline pkg/route/here_planner.go]: s/Bing/HERE

File: pkg/route/here_planner.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+func (p *herePlanner) TransitDistance(source *models.Address, destination *models.Address) (int, error) {
+
+	// Convert addresses to LatLong using geocode API. Do via goroutines and channel so we can do two
+	// requests in parallel
+	responses := make(chan addressLatLong)
+	var srcLatLong LatLong
+	var destLatLong LatLong
+	go p.getAddressLatLong(responses, source)
+	go p.getAddressLatLong(responses, destination)
+	for count := 0; count < 2; count++ {
```
Comment (macrael) [inline pkg/route/here_planner.go]: What happens if this response is killed by our global timeout? Will these goroutines be cleaned up?

File: pkg/route/planner_test.go [hunk 3]
```diff
@@ -18,6 +20,64 @@ type PlannerFullSuite struct {
...
+	StreetAddress1: "1000 Defense Pentagon",
+	City:           "Washington",
+	State:          "DC",
+	PostalCode:     "20301-1000"}
+
+// TestAddressPlanner is an expensive test which calls out to the Bing API.
+func (suite *PlannerFullSuite) TestAddressPlanner() {
+	distance, err := suite.planner.TransitDistance(&realAddressSource, &realAddressDestination)
+	if err != nil {
+		suite.T().Errorf("Failed to get distance from Bing - %v", err)
```
Comment (macrael) [inline pkg/route/planner_test.go]: s/Bing/Planner

File: pkg/route/here_planner.go (hunk 2 above)
Comment (ntwyman) [reply pkg/route/here_planner.go]: There is a timeout (15s) on the http request. If the request doesn't complete by then
```
p.httpClient.Get(query)
```
in the go routine will return an erro which the goroutine will drop in the `responses` chan and then end.

Even if the main request handling thread gets killed, all that will happen is that the goroutines will complete and then drop the results into channel that no-one listens to.

**Pull request merged**